src/app/loadout-builder/generated-sets/mod-utils.ts

```diff
@@ -0,0 +1,190 @@
+import { DimItem } from './../../inventory/item-types';
+import _ from 'lodash';
+import { LockableBuckets, ArmorSet, LockedArmor2ModMap, LockedArmor2Mod } from './../types';
+import { DestinyEnergyType } from 'bungie-api-ts/destiny2';
+import { getSpecialtySocketMetadata, Armor2ModPlugCategories } from 'app/utils/item-utils';
+
+const energyOrder = [
+  DestinyEnergyType.Void,
+  DestinyEnergyType.Thermal,
+  DestinyEnergyType.Arc,
+  DestinyEnergyType.Any
+];
+
+/**
+ * Checks that:
+ *   1. The armour piece is Armour 2.0
+ *   2. The mod matches the Armour energy OR the mod has the any Energy type
+ */
+const doEnergiesMatch = (mod: LockedArmor2Mod, item: DimItem) =>
+  item.isDestiny2() &&
+  item.energy &&
+  (mod.mod.plug.energyCost.energyType === DestinyEnergyType.Any ||
+    mod.mod.plug.energyCost.energyType === item.energy?.energyType);
+
+/**
+ * Assignes the general mods to armour pieces in assignments, including the energy specific ones
+ * i.e. Void Resist ect
+ */
+function assignGeneralMods(
+  setToMatch: readonly DimItem[],
+  generalMods: LockedArmor2Mod[],
+  assignments: Record<number, LockedArmor2Mod[]>
```

Comment:
Comment that this is mutated in the function

Reply:
Done